### DISCUSSION TITLE
FEATURE｜隐藏查找Java时的cmd窗口

### DISCUSSION THREAD

**Discussion by QianFuv** [MCSLTeam/MCSL2#80] [功能建议]
- [x] 并非电脑自身造成
- [x] MCSL 2 运行在受官方支持的平台上
___

### 一般要素
- 系统：win11
- 架构：amd64
- MCSL 2 版本：v2.1.3-230515-0005

### 问题描述
如标题

**Comment by lxhtt** [comment]
这是正常情况，在自动查找Java并获取Java信息
本可以直接获取文件版本，但由于无法跨平台，只能 `java -version` 来获取。
后期也许能够找到隐藏cmd的办法

**Comment by QianFuv** [comment]
> 这是正常情况，在自动查找Java并获取Java信息
> 本可以直接获取文件版本，但由于无法跨平台，只能 `java -version` 来获取。
> 后期也许能够找到隐藏cmd的办法

那也不需要隔几秒就寻找一次吧（加个按钮刷新不好吗

**Comment by lxhtt** [comment]
您所描述的并不符事实。cmd窗口不是为了搜索Java，而是为了获取搜索到的Java的版本信息。

**Comment by QianFuv** [comment]
> 您所描述的并不符事实。cmd窗口不是为了搜索Java，而是为了获取搜索到的Java的版本信息。

即便是这样也不需要两秒一次吧（？
有什么意义呢

**Comment by lxhtt** [comment]
我需要再次重新声明，cmd窗口是为了获取Java信息。我一次性全部给你执行需要多线程，但目前还没做，ok？至于两秒一次的说法，完全不存在。  
请自行反思为什么我语气如此强硬

**Comment by AresConnor** [comment]
> * [x]  并非电脑自身造成
> * [x]  MCSL 2 运行在受官方支持的平台上
> 
> ### 一般要素
> * 系统：win11
> * 架构：amd64
> * MCSL 2 版本：v2.1.3-230515-0005
> 
> ### 问题描述
> 如标题
> 
> ### 复现
> ### 可能的解决方案（？）

问题是因为在通过控制台输入java -version获取java版本的时候没有做cmd窗口隐藏，问题已经解决，将在下个构建中修复。至于多次弹框，是因为你的电脑上存在多个java

**Comment by QianFuv** [comment]
> > * [x]  并非电脑自身造成
> > * [x]  MCSL 2 运行在受官方支持的平台上
> > 
> > ### 一般要素
> > * 系统：win11
> > * 架构：amd64
> > * MCSL 2 版本：v2.1.3-230515-0005
> > 
> > ### 问题描述
> > 如标题
> > 
> > ### 复现
> > ### 可能的解决方案（？）
> 
> 问题是因为在通过控制台输入java -version获取java版本的时候没有做cmd窗口隐藏，问题已经解决，将在下个构建中修复。至于多次弹框，是因为你的电脑上存在多个java

好的谢谢解答

**Comment by QianFuv** [comment]
沟通不了一点 态度这样还是别做开发了
学学你的同事 找出问题而不是和我在这发火

**Comment by lxhtt** [comment]
您自行反思。我没有任何的义务重复申明一件正确的事。我所说的“隐藏cmd”，是指在获取搜索到的Java信息的过程中，因为打包后没有重定向控制台导致的闪现cmd。至于您说的所谓“2s搜索一次”，我最后说一次，程序不存在此逻辑。存在此逻辑早就被当作扫盘被报毒了。  
此外，您提到了“通过按钮刷新”的说法无处追溯。程序的逻辑如下：
 - 用户点下自动查找Java的按钮，程序启动子线程开始查找。  
 - 查找完毕后，通过子程序的方式，获取各个Java的版本信息。  
 - 获取后存入MCSL2文件夹中的配置。  
 - 刷新前端选择Java的列表，等待用户进入此页面。  
 - 用户选择，完毕。  
最后，我再次劝告，在同一个issue提出问题后，请着力解决这个问题，而不是提出毫不存在、完全离奇的其他令人窒息的东西。
